### PR TITLE
🐛 Detect progress regex from docker label ⬆️ 

### LINKED ISSUE #4376: Computational backend: service shall be able to define how progress is defined in the service
iSolve issues: 
Logs like the following are wrongly parsed as progress logs
```
1957728 / 1933878        (100.00% / 100.00%) : Total
1951940 / 1929102        ( 99.70% /  99.75%) : Dielectric
1951940 / 1933878        ( 99.70% / 100.00%) : Background
  16       16      32    1                 794.62
```

### PR DESCRIPTION
**Pull request by bisgaard-itis**
## What do these changes do?
- Allow a computational service to specify how its progress should be deduced from its log via a regexp pattern specifies in the labels. 

## Related issue/s
- fixes #4376

### CODE REVIEW

**Review by matusdrobuliak66 — APPROVED** [review]

File: packages/models-library/src/models_library/services.py [hunk 1]
```diff
@@ -494,6 +494,14 @@ class ServiceDockerData(ServiceKeyVersion, _BaseServiceCommonDataModel):
         regex=VERSION_RE,
         examples=["1.0.0"],
     )
+    progress_regexp: str | None = Field(
+        None,
+        alias="progress-regexp",
+        description="regexp pattern for detecting computational service's progress",
+        examples=[
+            "^(?:\\[?PROGRESS\\]?:?)?\\s*(?P<value>[0-1]?\\.\\d+|\\d+\\s*(?P<percent_sign>%))"
```
Comment (pcrespov) [inline packages/models-library/src/models_library/services.py:502]: Please rather add the example in `Config.schema_extra["examples"]` 
This way there is an automatic test that validates all examples there  (See packages/models-library/tests/test__models_examples.py)

File: services/dask-sidecar/requirements/_base.in [hunk 2]
```diff
@@ -15,6 +15,7 @@
 --requirement ../../../packages/models-library/requirements/_base.in
 --requirement ../../../packages/service-library/requirements/_base.in
 --requirement ../../../packages/settings-library/requirements/_base.in
+--requirement ../../../packages/service-integration/requirements/_base.in
```
Comment (pcrespov) [inline services/dask-sidecar/requirements/_base.in:18]: uff I would like to avoid this dependency if possible! Let me catch up with you offline.

File: services/dask-sidecar/requirements/_base.txt
Comment (pcrespov) [inline services/dask-sidecar/requirements/_base.txt:1]: careful full updates should be done in separate PRs

File: services/dask-sidecar/requirements/_base.in (hunk 2 above)
Comment (pcrespov) [reply services/dask-sidecar/requirements/_base.in:18]: I actually wonder why because I see no imports from service_integration. did you forget to version something?

**Review by pcrespov — CHANGES_REQUESTED** [review]
Thx.
Let's check if we can avoid OOIL dependency and let's avoid requirements updates. We can check that  together offline.

File: packages/models-library/src/models_library/services.py (hunk 1 above)
Comment (bisgaard-itis) [reply packages/models-library/src/models_library/services.py:502]: Done

File: services/dask-sidecar/requirements/_base.txt
Comment (bisgaard-itis) [reply services/dask-sidecar/requirements/_base.txt:1]: Yep, I reverted the inclusion of the OOIL into the dask sidecar, so now there are not dependency updates

**Review by pcrespov — APPROVED** [review]
:tada: nice!

File: services/dask-sidecar/src/simcore_service_dask_sidecar/computational_sidecar/core.py [hunk 3]
```diff
@@ -194,9 +195,15 @@ async def run(self, command: list[str]) -> TaskOutputData:
...
-            integration_version = await get_integration_version(
+            integration_version: version.Version = LEGACY_INTEGRATION_VERSION
+            progress_regexp: re.Pattern[str] = PROGRESS_REGEXP
+            if image_labels := await get_image_labels(
                 docker_client, self.docker_auth, self.service_key, self.service_version
-            )
+            ):
+                if iv := image_labels.integration_version:
+                    integration_version = version.Version(iv)
+                if pr := image_labels.progress_regexp:
```
Comment (GitHK) [inline services/dask-sidecar/src/simcore_service_dask_sidecar/computational_sidecar/core.py:205]: Q: can we have longer names here, please? I'm asking kindly to persuade you in changing them. :D

**Review by GitHK — APPROVED** [review]
👍

File: services/dask-sidecar/src/simcore_service_dask_sidecar/computational_sidecar/core.py (hunk 3 above)
Comment (bisgaard-itis) [reply services/dask-sidecar/src/simcore_service_dask_sidecar/computational_sidecar/core.py]: Done
